lib/Doctrine/DBAL/Driver/PDOPgSql/Driver.php

```diff
@@ -53,6 +53,14 @@ public function connect(array $params, $username = null, $password = null, array
                 $pdo->setAttribute(PDO::PGSQL_ATTR_DISABLE_PREPARES, true);
             }
 
+            /* defining client_encoding via SET NAMES to avoid inconsistent DSN support
+             * - the 'client_encoding' connection param only works with postgres >= 9.1
+             * - passing client_encoding via the 'options' param breaks pgbouncer support
+             */
+            if (isset($params['charset'])) {
+              $pdo->query('SET NAMES \''.$params['charset'].'\'');
```

Comment:
@deeky666 not too sure about that based on my tests.

I am running final tests now to get this all straightened out. The code that's already been merged WORKS atleast (doesn't break pgbouncer, or postgres). 

I'll work on _reliably_ detecting when we should pass client_encoding, or when we should use SET NAMES.

I'm not convinced using PGSQL_LIBPQ_VERSION is sufficent for the reasons I just explained. I am running some tests now to confirm this for sure.